packages/SwingSet/src/kernel/liveSlots.js

```diff
@@ -35,12 +36,13 @@ const DEFAULT_VIRTUAL_OBJECT_CACHE_SIZE = 3; // XXX ridiculously small value to
  * create a root object for the new vat The caller provided buildRootObject
  * function produces and returns the new vat's root object:
  *
- *     buildRootObject(vatPowers, vatParameters)
+ * buildRootObject(vatPowers, vatParameters)
  */
 function build(
   syscall,
   forVatID,
   cacheSize,
+  enableDisavow,
```

Comment:
Passing this enabling flag here seems weird to me.  Since the configuration flag that drives it is regulating whether something should be one of the vat powers, shouldn't that power just be present or not in the `vatPowers` object that's provided?

Reply:
There are two different flavors of `vatPowers` objects in play (and arguably they should have different names).

* the vat managers pass a `vatPowers` into `makeLiveSlots` which contains some non-vat-specific powers (`makeMarshal`, `transformTildot`, `testLog`), plus some vat-specific ones that don't rely upon anything in liveslots (`vatParameters`)
* liveslots builds a new `vatPowers` out of that one, plus things that rely upon the liveslots instance itself (mostly because they bottom out in syscalls): `D`, `exitVat`, `exitVatWithFailure`, `disavow`

We need liveslots to create our `disavow` function itself because `disavow` has to modify `valToSlot` and `slotToVal`, which are hidden inside liveslots. Plus it needs to invoke the `syscall` that was given to liveslots. Neither are available before `makeLiveSlots` is invoked, which is where the first flavor is provided.